dspace-api/src/main/java/org/dspace/discovery/indexobject/LDNMessageEntityIndexFactoryImpl.java

```diff
@@ -0,0 +1,156 @@
+    private LDNMessageService ldnMessageService;
+    @Autowired(required = true)
+    private ItemService itemService;
+
+    @Override
+    public Iterator<IndexableLDNNotification> findAll(Context context) throws SQLException {
+        final Iterator<LDNMessageEntity> ldnNotifications = ldnMessageService.findAll(context).iterator();
+        return new Iterator<>() {
+            @Override
+            public boolean hasNext() {
+                return ldnNotifications.hasNext();
+            }
+
+            @Override
+            public IndexableLDNNotification next() {
+                return new IndexableLDNNotification(ldnNotifications.next());
+            }
+        };
+    }
+
+    @Override
+    public String getType() {
+        return IndexableLDNNotification.TYPE;
+    }
+
+    @Override
+    public Optional<IndexableLDNNotification> findIndexableObject(Context context, String id) throws SQLException {
+        final LDNMessageEntity ldnMessage = ldnMessageService.find(context, id);
+        return ldnMessage == null ? Optional.empty() : Optional.of(new IndexableLDNNotification(ldnMessage));
+    }
+
+    @Override
+    public boolean supports(Object object) {
+        return object instanceof LDNMessageEntity;
+    }
+
+    @Override
+    public List<IndexableLDNNotification> getIndexableObjects(Context context, LDNMessageEntity object)
+        throws SQLException {
+        return Arrays.asList(new IndexableLDNNotification(object));
+    }
+
+    @Override
+    public SolrInputDocument buildDocument(Context context, IndexableLDNNotification indexableObject)
+        throws SQLException, IOException {
+        // Add the ID's, types and call the SolrServiceIndexPlugins
+        final SolrInputDocument doc = super.buildDocument(context, indexableObject);
+        final LDNMessageEntity ldnMessage = indexableObject.getIndexedObject();
+        // add schema, element, qualifier and full fieldName
+        doc.addField("notification_id", ldnMessage.getID());
+        doc.addField("queue_status_i", ldnMessage.getQueueStatus());
+        doc.addField("queue_status_s", LDNMessageEntity.getQueueStatus(ldnMessage));
+        addFacetIndex(doc, "queue_status", String.valueOf(ldnMessage.getQueueStatus()),
+            LDNMessageEntity.getQueueStatus(ldnMessage));
+        if (ldnMessage.getObject() == null || ldnMessage.getObject().getID() == null) {
+            throw new SQLException("ldnMessage " + ldnMessage.getID() + " is linked to no Item.");
+        }
+        Item item = itemService.findByIdOrLegacyId(context, ldnMessage.getObject().getID().toString());
+        if (item != null) {
+            addFacetIndex(doc, "object", item.getID().toString(), itemService.getMetadata(item, "dc.title"));
+            addFacetIndex(doc, "relateditem", item.getID().toString(), itemService.getMetadata(item, "dc.title"));
+        }
+        item = itemService.findByIdOrLegacyId(context, ldnMessage.getContext().getID().toString());
```

Comment:
I evidently have data where the Context can be null/empty.  I cannot reindex because this line throws a NullPointerException every time.  When this occurs it blocks all reindexing.  Before this line we need some sort of protection... this likely needs a check that both `ldn.getContext()` and `ldn.getContext().getID()` are *not null*.